pandas/_libs/hashtable_func_helper.pxi.in

```diff
@@ -141,32 +141,18 @@ cdef duplicated_{{dtype}}(const {{dtype}}_t[:] values, object keep='first', cons
     if keep not in ('last', 'first', False):
         raise ValueError('keep must be either "first", "last" or False')
 
-    if keep == 'last':
+    {{for cond, keep in [('if', '"last"'), ('elif', '"first"')]}}
+    {{cond}} keep == {{keep}}:
         {{if dtype == 'object'}}
         if True:
         {{else}}
         with nogil:
         {{endif}}
+            {{if keep == '"last"'}}
             for i in range(n - 1, -1, -1):
-                # equivalent: range(n)[::-1], which cython doesn't like in nogil
```

Comment:
is this comment no longer relevant?

Reply:
It is still relevant, but imo

```
range(n - 1, -1, -1)
```

is more clear than ``range(n)[::-1]`` anyway, so thought removing this would be fine.